packages/ui/src/Common.ts

```diff
@@ -1295,27 +1295,72 @@ export interface BodyProps {
 }
 
 export interface ButtonProps {
-  text: string;
-  // TODO make this work for all colors
-  color?: ButtonColor;
-  // default gray
-  disabled?: boolean; // default false
-  inline?: boolean; // default false
-  size?: "xs" | "sm" | "md" | "lg"; // default md
-  type?: "solid" | "ghost" | "outline"; // default solid
+  /**
+   * If true, the button will be disabled.
+   * @default false
+   */
+  disabled?: boolean;
+  /**
+   * If true, the button will take the full width of its container.
+   * @default false
+   */
+  fullWidth?: boolean;
+  /**
+   * The name of the icon to display in the button.
+   */
+  iconName?: IconName;
+  /**
+   * The position of the icon within the button.
+   * @default "left"
+   */
+  iconPosition?: "left" | "right";
+  /**
+   * If true, a loading spinner will be shown in the button.
+   */
   loading?: boolean;
-  onClick: any;
-  icon?: IconName;
-  iconColor?: IconColor;
-  withConfirmation?: boolean;
-  confirmationText?: string;
-  confirmationHeading?: string;
-  shape?: "rounded" | "pill";
+  /**
+   * The title of the confirmation modal.
+   * @default "Confirm"
+   */
+  modalTitle?: string;
+  /**
+   * The subtitle of the confirmation modal.
+   */
+  modalSubTitle?: string;
+  /**
+   * The text content of the confirmation modal.
+   * @default "Are you sure you want to continue?"
+   */
+  modalText?: string;
+  /**
+   * The test ID for the button, used for testing purposes.
+   */
   testID?: string;
-  tooltip?: {
-    text: string;
-    idealDirection?: TooltipDirection;
-  };
+  /**
+   * The text content of the button.
+   */
+  text: string;
+  /**
+   * The position of the tooltip.
+   */
+  tooltipPosition?: TooltipDirection;
+  /**
+   * The text content of the tooltip.
+   */
+  tooltipText?: string;
+  /**
+   * The type of the button, which determines its style.
+   * @default "primary"
+   */
+  variant?: "primary" | "secondary" | "muted" | "outline";
+  /**
+   * If true, a confirmation modal will be shown before the onClick action.
+   */
+  withConfirmationModal?: boolean;
+  /**
+   * The function to call when the button is clicked.
+   */
+  onClick: () => void;
```

Comment:
I think this needs to be void | Promise<void> to make the automatic loading work.